uaclient/clouds/azure.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+from urllib.error import HTTPError
+
+from uaclient.clouds import AutoAttachCloudInstance
+from uaclient import util
+
+
+IMDS_BASE_URL = "http://169.254.169.254/metadata/"
+IMDS_URLS = {
+    "pkcs7": IMDS_BASE_URL + "attested/document?api-version=2019-06-04",
+    "compute": IMDS_BASE_URL + "instance/compute?api-version=2019-06-04",
+}
+
+SYS_HYPERVISOR_PRODUCT_UUID = "/sys/hypervisor/uuid"
+DMI_CHASSIS_ASSET_TAG = "/sys/class/dmi/id/chassis_asset_tag"
+AZURE_CHASSIS_ASSET_TAG = "7783-7084-3265-9085-8269-3286-77"
+
+
+class UAAutoAttachAzureInstance(AutoAttachCloudInstance):
+
+    # mypy does not handle @property around inner decorators
+    # https://github.com/python/mypy/issues/1362
+    @property  # type: ignore
+    @util.retry(HTTPError, retry_sleeps=[1, 2, 5])
+    def identity_doc(self):
+        response = {}
```

Comment:
This is a nit for sure, but "response" in the context of the IMDS makes me assume that this will represent an HTTP response, whereas this is simply an internal data structure. Maybe even `s/response/responses/` would clear that up?